spec/models/mixins/inter_region_api_method_relay_spec.rb

```diff
@@ -0,0 +1,312 @@
+          args = {:my => "args", :here => 123}
+          expect(api_collection).to receive(action).with(args)
+          described_class.exec_api_call(region, collection_name, action, args)
+        end
+
+        it "defaults the args to an empty hash" do
+          expect(api_collection).to receive(action).with({})
+          described_class.exec_api_call(region, collection_name, action)
+        end
+
+        it "defaults the args to an empty hash when nil is explicitly passed as args" do
+          expect(api_collection).to receive(action).with({})
+          described_class.exec_api_call(region, collection_name, action, nil)
+        end
+      end
+
+      context "when a block is passed" do
+        let(:resource_proc) { -> { "some stuff" } }
+
+        it "calls the given action with the given args" do
+          expected_args = {:my => "args", :here => 123}
+          expect(api_collection).to receive(action) do |args, &block|
+            expect(args).to eq(expected_args)
+            expect(block.call).to eq("some stuff")
+          end
+          described_class.exec_api_call(region, collection_name, action, expected_args, &resource_proc)
+        end
+
+        it "defaults the args to an empty hash" do
+          expect(api_collection).to receive(action) do |args, &block|
+            expect(args).to eq({})
+            expect(block.call).to eq("some stuff")
+          end
+          described_class.exec_api_call(region, collection_name, action, &resource_proc)
+        end
+
+        it "defaults the args to an empty hash when nil is explicitly passed as args" do
+          expect(api_collection).to receive(action) do |args, &block|
+            expect(args).to eq({})
+            expect(block.call).to eq("some stuff")
+          end
+          described_class.exec_api_call(region, collection_name, action, nil, &resource_proc)
+        end
+      end
+    end
+  end
+
+  context "with an invalid class definition" do
+    describe "#api_relay_method" do
+      it "raises a NotImplementedError if the class does not have an api collection" do
+        expect {
+          Class.new do
+            extend InterRegionApiMethodRelay
+
+            def self.name
+              "RegionalMethodRelayTestClass"
+            end
+
+            def test_instance_method
+              "my test instance method"
+            end
+            api_relay_method :test_instance_method
+          end
+        }.to raise_error(NotImplementedError)
+      end
+    end
+
+    describe "#api_relay_class_method" do
+      it "raises a NotImplementedError if the class does not have an api collection" do
+        expect {
+          Class.new do
+            extend InterRegionApiMethodRelay
+
+            def self.name
+              "RegionalMethodRelayTestClass"
+            end
+
+            def self.test_instance_method
+              "my test instance method"
+            end
+            api_relay_class_method :test_instance_method do
```

Comment:
Note, I had look at this and prior test to figure out what was different.  I wonder if we can instantiate a shared class 'let' and class_eval the "difference" (api_relay_class_method vs. api_relay_method) so the test is more obvious.